File: maps/src/main/java/it/localhostsoftware/maps/UiSettings.java
package it.localhostsoftware.maps;

public interface UiSettings {
    void setZoomControlsEnabled(boolean var1);

    void setCompassEnabled(boolean var1);

    void setMyLocationButtonEnabled(boolean var1);

    void setIndoorLevelPickerEnabled(boolean var1);

    void setScrollGesturesEnabled(boolean var1);

    void setZoomGesturesEnabled(boolean var1);

    void setTiltGesturesEnabled(boolean var1);

    void setRotateGesturesEnabled(boolean var1);

    void setScrollGesturesEnabledDuringRotateOrZoom(boolean var1);

    void setAllGesturesEnabled(boolean var1);

    void setMapToolbarEnabled(boolean var1);

    boolean isZoomControlsEnabled();

    boolean isCompassEnabled();

    boolean isMyLocationButtonEnabled();

    boolean isIndoorLevelPickerEnabled();

    boolean isScrollGesturesEnabled();

    boolean isScrollGesturesEnabledDuringRotateOrZoom();

    boolean isZoomGesturesEnabled();

    boolean isTiltGesturesEnabled();

    boolean isRotateGesturesEnabled();

    boolean isMapToolbarEnabled();
}

File: maps/src/main/java/it/localhostsoftware/maps/UiSettings.kt
package it.localhostsoftware.maps

abstract class UiSettings<US>(val uiSettings: US) {
    abstract fun setAllGesturesEnabled(var1: Boolean)
    abstract var isZoomControlsEnabled: Boolean
    abstract var isCompassEnabled: Boolean
    abstract var isMyLocationButtonEnabled: Boolean
    abstract var isIndoorLevelPickerEnabled: Boolean
    abstract var isScrollGesturesEnabled: Boolean
    abstract var isScrollGesturesEnabledDuringRotateOrZoom: Boolean
    abstract var isZoomGesturesEnabled: Boolean
    abstract var isTiltGesturesEnabled: Boolean
    abstract var isRotateGesturesEnabled: Boolean
    abstract var isMapToolbarEnabled: Boolean
}